### PR TITLE
Update Github Actions for Python 3.11.0-rc.1

### PR DESCRIPTION
**Pull request by aminalaee**
If `3.11-dev` includes the RC release we can close this.

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11-dev"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11.0-rc.1"]
```
Comment (aminalaee) [inline .github/workflows/test-suite.yml:17]: Not really sure if `3.11-dev` will include `rc` releases or not. And the `3.11.0-rc.1` is not yet available in GH actions.

Comment (Kludex) [reply .github/workflows/test-suite.yml:17]: `-dev` includes all releases, not only alpha and beta.

Comment (Kludex) [reply .github/workflows/test-suite.yml:17]: ref.: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#python

Comment (aminalaee) [reply .github/workflows/test-suite.yml:17]: It's kind of ambiguous, it does mention alpha and beta but not RC. But I guess you're right.

**Pull request closed without merging**